### PR TITLE
tech-debt: Update connection-string to v2.0 and quaint to use upstream-mobc branch 

### PR DESCRIPTION
**Pull request by miguelff**
Reverts #3875, leading to the intended effects of #3833, but with a bug fixed

 One metric was not properly translated and thus missed from the output. Breaking a test in prisma/prisma. I added coverage for the missing metric and tested prisma/prisma with a published engine from this branch: "4.13.0-43.integration-mobc-upstream-30325ab26fa7d5669eb4b5c447f52a3918a4d9d7" to reproduce the problem.

### CODE REVIEW

**Review by tomhoule — APPROVED** [review]
lgtm

only thing that would be nice would be to upgrade tiberius to connection-string 0.2 too, but that's on a separate repo

File: Cargo.lock [hunk 1]
```diff
@@ -4631,7 +4639,7 @@ dependencies = [
  "byteorder",
  "bytes",
  "chrono",
- "connection-string",
+ "connection-string 0.1.13",
```
Comment (pimeys) [inline Cargo.lock:4642]: hum... should we update this one too?

File: libs/test-setup/Cargo.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
-connection-string = "0.1.10"
+connection-string = { workspace = true }
```
Comment (pimeys) [inline libs/test-setup/Cargo.toml:7]:
```suggestion
connection-string.workspace = true
```

File: query-engine/query-engine-node-api/Cargo.toml [hunk 3]
```diff
@@ -26,7 +26,7 @@ napi = { version = "2.12.4", default-features = false, features = ["napi8", "tok
 napi-derive = "2.12.3"
 
 thiserror = "1"
-connection-string = "0.1"
+connection-string = { workspace = true }
```
Comment (pimeys) [inline query-engine/query-engine-node-api/Cargo.toml:29]:
```suggestion
connection-string.workspace = true
```

File: query-engine/query-engine/src/server/mod.rs [hunk 4]
```diff
@@ -239,16 +238,7 @@ fn playground_handler() -> Response<Body> {
...
-        if query.contains("format=json") {
-            MetricFormat::Json
-        } else {
-            MetricFormat::Prometheus
-        }
-    } else {
-        MetricFormat::Prometheus
-    };
-
+    let requested_json = req.uri().query().map(|q| q.contains("format=json")).unwrap_or_default();
```
Comment (pimeys) [inline query-engine/query-engine/src/server/mod.rs:241]: better

File: schema-engine/cli/Cargo.toml [hunk 5]
```diff
@@ -23,8 +23,8 @@ tempfile = "3.1.0"
 test-setup = { path = "../../libs/test-setup" }
 test-macros = { path = "../../libs/test-macros" }
 url = "2.1.1"
-connection-string = "0.1"
 indoc.workspace = true
+connection-string = { workspace = true }
```
Comment (pimeys) [inline schema-engine/cli/Cargo.toml:27]:
```suggestion
connection-string.workspace = true
```

File: schema-engine/connectors/sql-schema-connector/Cargo.toml [hunk 6]
```diff
@@ -22,7 +22,7 @@ sql-ddl = { path = "../../../libs/sql-ddl" }
 user-facing-errors = { path = "../../../libs/user-facing-errors", features = ["sql"] }
 
 chrono = { version = "0.4" }
-connection-string = "0.1.10"
+connection-string = { workspace = true }
```
Comment (pimeys) [inline schema-engine/connectors/sql-schema-connector/Cargo.toml:25]:
```suggestion
connection-string.workspace = true
```

File: schema-engine/sql-introspection-tests/Cargo.toml [hunk 7]
```diff
@@ -13,7 +13,7 @@ user-facing-errors = { path = "../../libs/user-facing-errors" }
 test-setup = { path = "../../libs/test-setup" }
 
 enumflags2 = "0.7"
-connection-string = "0.1"
+connection-string = { workspace = true }
```
Comment (pimeys) [inline schema-engine/sql-introspection-tests/Cargo.toml:16]:
```suggestion
connection-string.workspace = true
```

File: schema-engine/sql-migration-tests/Cargo.toml [hunk 8]
```diff
@@ -16,7 +16,7 @@ prisma-value = { path = "../../libs/prisma-value" }
 bigdecimal = "0.3"
 chrono = "0.4.15"
 colored = "2"
-connection-string = "0.1.13"
+connection-string = { workspace = true }
```
Comment (pimeys) [inline schema-engine/sql-migration-tests/Cargo.toml:19]:
```suggestion
connection-string.workspace = true
```

**Review by pimeys — COMMENTED** [review]
A few nitpicks, and I think something is still using connection-string 0.1.x...

**Comment by miguelff** [comment]
I will update tiberius to connection-string 0.2 in a different PR. Thanks for the feedback

**Comment by miguelff** [comment]
I finally updated tiberius in this same branch after releasing a new version of the library. Re: https://github.com/prisma/tiberius/pull/286 and https://github.com/prisma/tiberius/pull/288

Waiting for CI feedback on prisma/prisma and I will merge.

**Comment by miguelff** [comment]
Metrics tests are passing on prisma side. Merging.

**Pull request merged**